source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -463,27 +463,33 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
 
   parsed_alpn_protocols_ = parseAlpnProtocols(config.alpnProtocols());
 
-  // To enumerate the required builtin ciphers, curves, algorithms, and
-  // versions, uncomment '#define LOG_BUILTIN_STAT_NAMES' below, and run
-  //  bazel test //test/extensions/transport_sockets/tls/... --test_output=streamed
-  //      | grep " Builtin ssl." | sort | uniq
-  // #define LOG_BUILTIN_STAT_NAMES
-  //
-  // TODO(#8035): improve tooling to find any other built-ins needed to avoid
-  // contention.
-
-  // Ciphers
-  stat_name_set_->rememberBuiltin("AEAD-AES128-GCM-SHA256");
-  stat_name_set_->rememberBuiltin("ECDHE-ECDSA-AES128-GCM-SHA256");
-  stat_name_set_->rememberBuiltin("ECDHE-RSA-AES128-GCM-SHA256");
-  stat_name_set_->rememberBuiltin("ECDHE-RSA-AES128-SHA");
-  stat_name_set_->rememberBuiltin("ECDHE-RSA-CHACHA20-POLY1305");
+  // Ciphers are configured as a string delimited by ":", with equivalence
+  // groups given as "[alt1|alt2]", and exclusions preceded by "!". For the
+  // purposes of collecting stats -- we want to track all these names. We don't
+  // need to fully parse out the structure of the cipher suites -- just extract
+  // out the names. We skip exclusions as those do not show up as entire
+  // stat-name segments, so they would not match.
+  for (absl::string_view cipher_suite :
+       absl::StrSplit(config.cipherSuites(), absl::ByAnyChar(":|[]"))) {
+    if (!cipher_suite.empty() && cipher_suite[0] != '!') { // skip exclusions.
+      stat_name_set_->rememberBuiltin(cipher_suite);
+    }
+  }
+
+  // This cipher is referenced in a test, though it's not obvious how or why.
   stat_name_set_->rememberBuiltin("TLS_AES_128_GCM_SHA256");
```

Comment:
Which test? Is this something we can/should fix?

Reply:
I added documentation about what tests caused us to try to record values on stats that were not recorded from the cipher-suites config above. Turns out I missed one, so I added that.

It's probably worth digging into why those names showed up, but probably that should be a follow-up, and would need an SSL expert's help.